### PR TITLE
IX Bid Adapter: retrieve user/agent hints and fix tmax issue

### PR DESCRIPTION
**Pull request by ccorbo**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Few small changes here:

- Retrieve UserAgent Client Hints in PBJS Adapter
- Fix tmax issue outlined here: https://github.com/prebid/Prebid.js/issues/9287
- Change mtype logic

### CODE REVIEW

File: modules/ixBidAdapter.js [hunk 1]
```diff
@@ -629,7 +628,7 @@ function buildRequest(validBidRequests, bidderRequest, impressions, version) {
   }
 
   const r = {};
-  const tmax = config.getConfig('bidderTimeout');
+  const tmax = deepAccess(bidderRequest, 'timeout', config.getConfig('bidderTimeout'));
```
[the remark this answers is not in the document]
Comment (oronno) [reply modules/ixBidAdapter.js:631]: if `bidderRequest` doesn't have a `timeout` value defined, then global timeout will be used as default which we get from config `config.getConfig('bidderTimeout')`

[the remark this answers is not in the document]
Comment (patmmccann) [reply modules/ixBidAdapter.js:631]: Just write a unit test to ensure it always exists. There is no need to fallback.

[the remark this answers is not in the document]
Comment (oronno) [reply modules/ixBidAdapter.js:631]: > There is no need to fallback.

Not sure if I get your correctly.
We are setting **tmax** value from (in order):
1) bidderRequest
2) global config 

It's not a mandatory field in bidderRequest. So if bidderRequest don't contain it, tmax will be value from global config.
Above code cover both cases.  This fixes reported tmax bug https://github.com/prebid/Prebid.js/issues/9287  
Note, this PR also cover overriding logic by a test `should set ixdiag.tmax value from bidderRequest overriding global config bidderTimeout`

[the remark this answers is not in the document]
Comment (patmmccann) [reply modules/ixBidAdapter.js:631]: bidderRequest will always contain it, correct @dgirardi ?

[the remark this answers is not in the document]
Comment (dgirardi) [reply modules/ixBidAdapter.js:631]: correct, `bidderRequest.timeout` is [always there](https://github.com/prebid/Prebid.js/blob/d7669688a570a2b57904dd600525e9f3d5b6cdca/src/adapterManager.js#L299) and the fallback is redundant.

**Pull request merged**